ethereum/core/src/main/java/org/hyperledger/besu/ethereum/proof/WorldStateProofProvider.java

```diff
@@ -92,4 +109,81 @@ public WorldStateProofProvider(final WorldStateStorage worldStateStorage) {
         b -> b,
         b -> b);
   }
+
+  public boolean isValidRangeProof(
+      final Bytes32 startKeyHash,
+      final Bytes32 endKeyHash,
+      final Bytes32 rootHash,
+      final List<Bytes> proofs,
+      final TreeMap<Bytes32, Bytes> keys) {
+
+    // check if it's monotonic increasing
+    if (!Ordering.natural().isOrdered(keys.keySet())) {
+      return false;
+    }
+
+    // when proof is empty we need to have all the keys to reconstruct the trie
+    if (proofs.isEmpty()) {
+      final MerklePatriciaTrie<Bytes, Bytes> trie =
+          new SimpleMerklePatriciaTrie<>(Function.identity());
+      // add the received keys in the trie
+      for (Map.Entry<Bytes32, Bytes> key : keys.entrySet()) {
+        trie.put(key.getKey(), key.getValue());
+      }
+
+      return rootHash.equals(trie.getRootHash());
+    }
+
+    // reconstruct a part of the trie with the proof
+    Map<Bytes32, Bytes> proofsEntries = Collections.synchronizedMap(new HashMap<>());
```

Comment:
Does this need to be a synchronizedMap?  I am not seeing async usage but maybe I am missing it.

Reply:
Yes still related to an old iteration and forgot to update it